Fix for non-ascii tex
--- a/parsers/tex.c
+++ b/parsers/tex.c
@@ -34,9 +34,10 @@
  */
 #define isType(token,t)		(bool) ((token)->type == (t))
 #define isKeyword(token,k)	(bool) ((token)->keyword == (k))
+#define isAscii(c) ((c >= 0) && (c < 0x80))
 #define isIdentChar(c) \
-	(isalpha (c) || isdigit (c) || (c) == '$' || \
-		(c) == '_' || (c) == '#' || (c) == '-' || (c) == '.' || (c) == ':')
+	((isAscii(c) && (isalpha (c) || isdigit (c) || (c) == '$' || \
+		(c) == '_' || (c) == '#' || (c) == '-' || (c) == '.' || (c) == ':')) || !isAscii(c))
 
 /*
  *	 DATA DECLARATIONS
